src/asm_cfg.cpp

```diff
@@ -216,12 +224,20 @@ std::map<std::string, int> collect_stats(const cfg_t& cfg) {
 }
 
 cfg_t prepare_cfg(const InstructionSeq& prog, const program_info& info, bool simplify) {
+    // Convert the instruction sequence to a deterministic control-flow graph.
     cfg_t det_cfg = instruction_seq_to_cfg(prog);
+
+    // Annotate the CFG by adding in assertions before every memory instruction.
     explicate_assertions(det_cfg, info);
+
+    // Translate conditional jumps to non-determinstic jumps.
     cfg_t cfg = to_nondet(det_cfg);
 
+    // Except when debugging, combine basic blocks where possible,
+    // i.e., where there is no jump to a later basic block.
```

Comment:
I think it is more precise to say something like "combine chains of instructions to basic blocks" or something of the sort. The single instructions _can_ be thought of as trivial basic blocks, but the reader probably just sees them as instructions at this point.

Also, this is not exactly about jumps. A basic block is a sequence of instructions with a single entry point and a single exit point. Unconditional jumps do not necessarily break basic blocks, and fallthrough may or may not be concatenated - it depends on whether there exist a jump somewhere with the following instruction as its target.

Reply:
Updated